src/amo/components/SearchFilters/index.js

```diff
@@ -79,6 +84,11 @@ export class SearchFiltersBase extends React.Component {
       newFilters.featured = true;
     }
 
+    if (newFilters.page) {
+      // We reset the page as filters have changed.
+      newFilters.page = 1;
+    }
+
```

Comment:
At some point, this and the `router.push` could be a helper (or extracted into a new method). Worth doing there or?

Reply:
> At some point, this and the router.push could be a helper (or extracted into a new method). Worth doing there or?

Yeah, I'd say you can share it in component method:

```js
doSearch({ newFilters, filtersWereChanged = false }) {
  const { ... } = this.props;

  if (filtersWereChanged && newFilters.page) {
    // Since it's now a new search, reset the page.
    newFilters.page = 1;
  }

  router.push({
      pathname: `/${lang}/${clientApp}${pathname}`,
      query: convertFiltersToQueryParams(newFilters),
  });
}
```